### ISSUE TITLE
Unknown field name 'message_retention_duration' in initialization map entry

### ISSUE THREAD

**Issue by selivandex** [googleapis/google-cloud-ruby#15692]
#### Environment details

  - OS: MacOS
  - Ruby version: 2.7.3
  - Gem name and version: google-cloud-pubsub - 2.8.1 AND 2.9.0

#### Steps to reproduce

  1. ...

#### Code example

```ruby
pubsub = Google::Cloud::PubSub.new
pubsub.create_topic('topic-name')
```

#### Full backtrace

```sh
ArgumentError: Unknown field name 'message_retention_duration' in initialization map entry.
from /Users/user/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/gapic-common-0.3.4/lib/gapic/protobuf.rb:38:in `initialize'
```

**Comment by quartzmo** [comment]
@selivandex I am unable to reproduce this error locally, but I'll try to investigate the cause through static code review. Please provide any additional details you think might help, like the output of `bundle update`.

```
quartzmo@Chriss-iMac google-cloud-pubsub % ruby -v
ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x86_64-darwin19]
quartzmo@Chriss-iMac google-cloud-pubsub % bundle exec rake console
irb(main):001:0> require "google/cloud/pubsub"
=> true
irb(main):002:0> Google::Cloud::PubSub::VERSION
=> "2.9.0"
irb(main):003:0> pubsub = Google::Cloud::PubSub.new
=> #<Google::Cloud::PubSub::Project:0x00007fe42aa30b40 @service=#<Google::Cloud::PubSub::Service (...)>>
irb(main):004:0> pubsub.create_topic('topic-name')
=> #<Google::Cloud::PubSub::Topic:0x00007fe42a9bb8b8 @service=#<Google::Cloud::PubSub::Service (...)>, @grpc=<Google::Cloud::PubSub::V1::Topic: name: "projects/.../topics/topic-name", labels: {}, kms_key_name: "", satisfies_pzs: false>, @resource_name=nil, @exists=nil, @async_opts={}>
irb(main):005:0> 
```

**Comment by quartzmo** [comment]
I appears from your backtrace that you are using `gapic-common-0.3.4`, which is out of date. The current version is [0.7.0](https://rubygems.org/gems/gapic-common/versions). Try running `bundle update` to install the latest dependencies for google-cloud-pubsub.

**Comment by selivandex** [comment]
> I appears from your backtrace that you are using `gapic-common-0.3.4`, which is out of date. The current version is [0.7.0](https://rubygems.org/gems/gapic-common/versions). Try running `bundle update` to install the latest dependencies for google-cloud-pubsub.

Really. After update it works!

**Comment by bmyers427** [comment]
I'm getting a similar issue and updating `gapic-common` didn't resolve it. I'm using Ruby 2.5.8, `google-cloud-pubsub-2.9.1` and `gapic-common-0.8.0`. Creating a topic throws the following error:

`irb(main):005:0> pubsub.create_topic('test-topic')
Traceback (most recent call last):
       16: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/cli.rb:27:in `dispatch'
       15: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
       14: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
       13: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
       12: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/cli.rb:463:in `exec'
       11: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:28:in `run'
       10: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:74:in `kernel_load'
        9: from /usr/local/lib/ruby/site_ruby/2.5.0/bundler/cli/exec.rb:74:in `load'
        8: from /usr/local/bin/irb:11:in `<top (required)>'
        7: from (irb):5
        6: from /usr/local/bundle/gems/google-cloud-pubsub-2.9.1/lib/google/cloud/pubsub/project.rb:267:in `create_topic'
        5: from /usr/local/bundle/gems/google-cloud-pubsub-2.9.1/lib/google/cloud/pubsub/service.rb:148:in `create_topic'
        4: from /usr/local/bundle/gems/google-cloud-pubsub-v1-0.4.0/lib/google/cloud/pubsub/v1/publisher/client.rb:281:in `create_topic'
        3: from /usr/local/bundle/gems/gapic-common-0.8.0/lib/gapic/protobuf.rb:38:in `coerce'
        2: from /usr/local/bundle/gems/gapic-common-0.8.0/lib/gapic/protobuf.rb:38:in `new'
        1: from /usr/local/bundle/gems/gapic-common-0.8.0/lib/gapic/protobuf.rb:38:in `initialize'
ArgumentError (Unknown field name 'message_retention_duration' in initialization map entry.)`

**Comment by coiti** [comment]
I found that this issue is actually caused by having an older version of `google-cloud-pubsub-v1` installed.

`message_retention_duration` was implemented in version 0.6.0 of that gem ([see changelog](https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-pubsub-v1/CHANGELOG.md#060--2021-08-11)).

The solution was to bump `google-cloud-pubsub-v1` to `>= 0.6.0`.

**Comment by bmyers427** [comment]
thanks! I was able to get it working after upgrading `google-cloud-pubsub-v1` to `0.7.0`.
